ui-tests/src/test/java/io/jenkins/plugins/analysis/warnings/WarningsPluginUiTest.java

```diff
@@ -0,0 +1,611 @@
+        assertThat(analysisSummary).isDisplayed()
+                .hasTitleText("Static Analysis: 25 warnings")
+                .hasAggregation("FindBugs, CPD, CheckStyle, PMD")
+                .hasNewSize(23)
+                .hasFixedSize(2)
+                .hasReferenceBuild(1);
+
+        AnalysisResult result = analysisSummary.openOverallResult();
+        assertThat(result).hasActiveTab(Tab.TOOLS).hasTotal(25)
+                .hasOnlyAvailableTabs(Tab.TOOLS, Tab.PACKAGES, Tab.FILES, Tab.CATEGORIES, Tab.TYPES, Tab.ISSUES);
+    }
+
+    private IssuesRecorder addRecorder(final FreeStyleJob job) {
+        return job.addPublisher(IssuesRecorder.class, recorder -> {
+            recorder.setTool("CheckStyle");
+            recorder.addTool("FindBugs");
+            recorder.addTool("PMD");
+            recorder.addTool("CPD",
+                    cpd -> cpd.setHighThreshold(8).setNormalThreshold(3));
+            recorder.setEnabledForFailure(true);
+        });
+    }
+
+    /**
+     * Test to check that the issue filter can be configured and is applied.
+     */
+    @Test
+    public void shouldFilterIssuesByIncludeAndExcludeFilters() {
+        FreeStyleJob job = createFreeStyleJob("issue_filter/checkstyle-result.xml");
+        job.addPublisher(IssuesRecorder.class, recorder -> {
+            recorder.setTool("CheckStyle");
+            recorder.setEnabledForFailure(true);
+            recorder.addIssueFilter("Exclude categories", "Checks");
+            recorder.addIssueFilter("Include types", "JavadocMethodCheck");
+        });
+
+        job.save();
+
+        Build build = buildJob(job);
+
+        assertThat(build.getConsole()).contains(
+                "Applying 2 filters on the set of 4 issues (3 issues have been removed, 1 issues will be published)");
+
+        AnalysisResult resultPage = new AnalysisResult(build, "checkstyle");
+        resultPage.open();
+
+        IssuesTable issuesTable = resultPage.openIssuesTable();
+        assertThat(issuesTable).hasSize(1);
+    }
+
+    private void reconfigureJobWithResource(final FreeStyleJob job, final String fileName) {
+        job.configure(() -> job.copyResource(WARNINGS_PLUGIN_PREFIX + fileName));
+    }
+
+    /**
+     * Creates and builds a maven job and verifies that all warnings are shown in the summary and details views.
+     */
+    @Test
+    @WithPlugins({"maven-plugin", "analysis-model-api@7.0.4"})
+    public void shouldShowMavenWarningsInMavenProject() {
+        MavenModuleSet job = createMavenProject();
+        copyResourceFilesToWorkspace(job, SOURCE_VIEW_FOLDER + "pom.xml");
+
+        IssuesRecorder recorder = job.addPublisher(IssuesRecorder.class);
+        recorder.setToolWithPattern("Maven", "");
+        recorder.setEnabledForFailure(true);
+
+        job.save();
+
+        Build build = buildJob(job).shouldSucceed();
+
+        System.out.println("-------------- Console Log ----------------");
+        System.out.println(build.getConsole());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [System.out.println is used](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=5151082)